scripts/deno/createMainChangelog.ts

```diff
@@ -0,0 +1,207 @@
+import { readFileSync, writeFileSync } from "node:fs";
+import { List, ListItem, Root, Text } from "npm:@types/mdast";
+import { Node } from "npm:@types/unist";
+import { heading, root, text } from "npm:mdast-builder";
+import remarkParse from "npm:remark-parse";
+import remarkStringify from "npm:remark-stringify";
+import { unified } from "npm:unified";
+import { EXIT, SKIP, visit } from "npm:unist-util-visit";
+import { visitParents } from "npm:unist-util-visit-parents";
+import { getChangelogs } from "./utils.ts";
+
+/**
+ * Small diagram of the process:
+
+           original 
+        markdown files
+               |
+               V
+          custom JSON 
+        representation
+         of changelog
+               |
+               V
+        final markdown
+ */
+
+type PackageName = string;
+type Version = string;
+type RawASTNode = Node;
+type SemverChangeHeading = Record<string, RawASTNode[]>;
+type VersionEntry = Record<PackageName, SemverChangeHeading>;
+type Changelog = Record<Version, VersionEntry>;
+
+const upsertEntry = (
+  changelog: Changelog,
+  {
+    lastSeenPackage,
+    lastSeenVersion,
+    lastSeenSemverHeading,
+  }: {
+    lastSeenPackage: string;
+    lastSeenVersion: string;
+    lastSeenSemverHeading: string;
+  },
+  value: RawASTNode
+) => {
+  if (!changelog[lastSeenVersion]) {
+    changelog[lastSeenVersion] = {};
+  }
+  if (!changelog[lastSeenVersion][lastSeenPackage]) {
+    changelog[lastSeenVersion][lastSeenPackage] = {};
+  }
+  const raw_ast_nodes =
+    changelog[lastSeenVersion][lastSeenPackage][lastSeenSemverHeading];
+  if (!raw_ast_nodes) {
+    changelog[lastSeenVersion][lastSeenPackage][lastSeenSemverHeading] = [
+      value,
+    ];
+  } else {
+    raw_ast_nodes.push(value);
+  }
+
+  const updatedVersion = changelog[lastSeenVersion] || [];
+  changelog[lastSeenVersion] = updatedVersion;
```

Comment:
Bare en liten forkorting egentlig, men arrester meg gjerne! Jeg er ikke helt trygg på koden, så godt mulig jeg misforstår noe

```
  changelog[lastSeenVersion] ||= {};
  changelog[lastSeenVersion][lastSeenPackage] ||= {};

  const raw_ast_nodes = changelog[lastSeenVersion][lastSeenPackage][lastSeenSemverHeading];
  changelog[lastSeenVersion][lastSeenPackage][lastSeenSemverHeading] = raw_ast_nodes ? [...raw_ast_nodes, value] : [value];
```
Kort forklaring:
`||=` assigner _hvis_ f.eks. `changelog[lastSeenVersion]` er nullish.

If-else gjorde jeg til en one-liner. Siden det er en form for assignment i begge tilfeller, så er det kanskje ved øyekast lettere å forstå hva som skjer. Og det er kortere da ;) 

`const updatedVersion` tok jeg bort, men hvis det er behov kan jo nullish assignment funke her også!